src/Raven.Server/Documents/TransactionOperationsMerger.cs

```diff
@@ -848,5 +1092,173 @@ public void Dispose()
                 result.TaskCompletionSource.TrySetCanceled();
             }
         }
+
+
+        public enum TxInstruction
+        {
+            BeginTx,
+            Commit,
+            Rollback,
+            DisposeTx,
+            FinishAsyncAndStartNew,
+
+            //Todo I think this is not relevant all the async instruction can run as one unit in  FinishAsyncAndStartNew
+            EndAsyncCommit
+        }
+
+        private RecordingState _recordingStatus = null;
+
+        public void StartRecording(string filePath)
+        {
+            if (_recordingStatus != null)
+                return;
+            Interlocked.CompareExchange(ref _recordingStatus, new RecordingState.BeforeEnabledRecordingState(this, filePath), null);
+            _waitHandle.Set();
+        }
+
+        public void StopRecording()
+        {
+            var recordingStatus = _recordingStatus;
+            if (recordingStatus != null)
+            {
+                recordingStatus.Shutdown();
+                _waitHandle.Set();
+            }
+        }
+
+        public void Replay(string filePath)
```

Comment:
This needs to accept a Stream, not a file path

Reply:
done

Reply:
done